### ISSUE TITLE
Document the container/app lifecycle

### ISSUE THREAD

**Issue by shaunmulligan** [resin-io/docs#295]
Users want to know things like:
* what signals are sent to the containers on kill. 
* how does handover between updates happen by default
* what happens when a container crashes
* what happens when a container hangs
* is there an automatic roll back?

**Comment by imrehg** [comment]
A signal handling example (but need to check if still works with current setup because it's a few months old): https://github.com/resin-io-projects/resin-watchdog/blob/master/src/start.sh

**Comment by shaunmulligan** [comment]
Yeah I think that's gonna change in the next release, so probably won't work anymore :/
